### PR TITLE
Finished API Routes and fixed according to feedback

### PR DESCRIPTION
**Pull request by lizzkats**

### CODE REVIEW

File: test/database/queries/questions_test.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import chai, { expect } from 'chai'
 import * as question from '../../../src/database/queries/questions'
 
-describe('Question Tests', () => {
+describe.only('Question Tests', () => {
```
Comment (punitrathore) [inline test/database/queries/questions_test.js]: you probably didnt want this to be commited :)

File: app.js [hunk 2]
```diff
@@ -1,9 +1,97 @@
...
 
-app.use(express.static(__dirname + '/public/dist/'));
-app.use(express.static(__dirname + '/src/browser/'));
+app.use( bodyParser.json() )
+app.use( bodyParser.urlencoded( {extended: false}) )
+app.use( express.static(__dirname + '/public/dist/') );
+app.use( express.static(__dirname + '/src/browser/') );
+app.use( ( cors() )
+
+app.get('/users/:name', (request, response) => {
```
Comment (punitrathore) [inline app.js]: please create a new file called `src/server/routes/users.js`, and put the user routes in there

File: app.js [hunk 3]
```diff
@@ -1,9 +1,97 @@
...
+})
+
+app.get('/users/:github_handle', (request, response) => {
+  const { github_handle } = request.params
+  users.findbyGithub(github_handle)
+  .then( results => response.json( results ) )
+  .catch( err => console.log('err', err) )
+})
+
+app.get('/questions/:tags', (request, response) => {
```
Comment (punitrathore) [inline app.js]: same for questions too 

please create a new file called `src/server/routes/questions.js`, and put the question routes in there.

And these routes should have a prefix `api/users`

for example 
  - `/api/question/:tags`
  - `/api/question/:level`

Comment (punitrathore) [inline app.js]: also we want a route - `/api/questions` which accepts the  GET request method, and takes 2 query params - `topic` & `difficulty`.

**Comment by lizzkats** [comment]
commit bb6c5d2 is the final commit and has some issues with the questions route

File: test/server/routes/questions_test.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      .end((err, res) => {
+        res.should.have.status(200);
+        res.body.should.be.a('array');
+        res.body.length.should.be.eql(0);
+        done();
+      })
+  })
+  it('Should respond with a status code of 200 and get by id', (done) => {
+    chai.request('http://interview.learnersguild.dev/api')
+    .get('/:id')
```
Comment (punitrathore) [inline test/server/routes/questions_test.js]: this needs to be an actual `id`. Also the request url seems to be off. it should be `http://interview.learnersguild.dev/api/questions/1`

File: test/server/routes/questions_test.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      .end((err, res) => {
+        res.should.have.status(200);
+        res.body.should.be.a('object');
+        done();
+      })
+  })
+  it('Should respond with a status code of 200 and get by id', (done) => {
+    chai.request('http://interview.learnersguild.dev/api')
+    .get('/:level')
```
Comment (punitrathore) [inline test/server/routes/questions_test.js]: same re:level. `level` needs to be an actual value

**Pull request merged**